### PR TITLE
Allow closure for contentOptions

### PR DESCRIPTION
**Pull request by bscheshirwork**
## Scope
This pull request includes a

- [x] Bug fix
- [ ] New feature
- [ ] Translation

## Changes
The following changes were made (this change is also documented in the [change log](https://github.com/kartik-v/yii2-export/blob/master/CHANGE.md)):

- Can use closure for contentOptions

## Related Issues
If this is related to an existing ticket, include a link to it as well.
#328

### CODE REVIEW

File: src/ExportMenu.php [hunk 1]
```diff
@@ -1391,7 +1391,11 @@ public function generateRow($model, $key, $index)
             } else {
                 $value = '';
             }
-            $format = ArrayHelper::getValue($column->contentOptions, 'cellFormat', null);
+            $contentOptions = $column->contentOptions;
+            if (is_callable($contentOptions)) {
+                $contentOptions = $contentOptions($model, $key, $index);
```
Comment (awwar) [inline src/ExportMenu.php:1396]: I'm not sure, but the closure seems to have the following signature:
`function ($model, $key, $index, $column)`

Is it possible to implement it?

Comment (awwar) [reply src/ExportMenu.php:1396]: A reference: https://github.com/kartik-v/yii2-grid/blob/master/src/ActionColumn.php#L120

Comment (udokmeci) [reply src/ExportMenu.php:1396]: @bscheshirwork  Thank you for patch but @awwar is right. It requires exactly 4 columns :D .

Comment (bscheshirwork) [reply src/ExportMenu.php]: Thanks! It is correct now?

Comment (awwar) [reply src/ExportMenu.php]: Yep, now it's valid! 
But let make condition more strict. Change it from `if (is_callable($contentOptions)) {` to `if ($contentOptions instanceof Closure) {`.

Comment (bscheshirwork) [reply src/ExportMenu.php]: how about something like `[classname, methodname]`? this is callable, but not \Closure

Comment (awwar) [reply src/ExportMenu.php]: Yeah, it's look like good solution and it will work, but!
In kartik docs - https://github.com/kartik-v/yii2-grid/blob/master/src/ActionColumn.php#L118 `contentOptions` can be `array` or `\Closure`.
In yii2 docs - https://github.com/yiisoft/yii2/blob/master/framework/grid/Column.php#L105 condition looks like this.
So, if u pass obj with  `__invoke()` magic method (https://www.php.net/manual/ru/language.oop5.magic.php#object.invoke) it's will work on export but do not work in view.

In short, let's focus on the documentation :)

But if you deem it necessary to leave it this way, ok, let's leave it this way.
